cmd/promtool/unittest.go

```diff
@@ -312,30 +313,18 @@ func (tg *testGroup) test(evalInterval time.Duration, groupOrderMap map[string]i
 					})
 				}
 
-				var sb strings.Builder
-				if gotAlerts.Len() != expAlerts.Len() {
-					if tg.TestGroupName != "" {
-						fmt.Fprintf(&sb, "    name: %s,\n", tg.TestGroupName)
-					}
-					fmt.Fprintf(&sb, "    alertname:%s, time:%s, \n", testcase.Alertname, testcase.EvalTime.String())
-					fmt.Fprintf(&sb, "        exp:%#v, \n", expAlerts.String())
-					fmt.Fprintf(&sb, "        got:%#v", gotAlerts.String())
-
-					errs = append(errs, errors.New(sb.String()))
-				} else {
-					sort.Sort(gotAlerts)
-					sort.Sort(expAlerts)
-
-					if !reflect.DeepEqual(expAlerts, gotAlerts) {
-						if tg.TestGroupName != "" {
-							fmt.Fprintf(&sb, "    name: %s,\n", tg.TestGroupName)
-						}
-						fmt.Fprintf(&sb, "    alertname:%s, time:%s, \n", testcase.Alertname, testcase.EvalTime.String())
-						fmt.Fprintf(&sb, "        exp:%#v, \n", expAlerts.String())
-						fmt.Fprintf(&sb, "        got:%#v", gotAlerts.String())
+				sort.Sort(gotAlerts)
+				sort.Sort(expAlerts)
 
-						errs = append(errs, errors.New(sb.String()))
+				if !reflect.DeepEqual(expAlerts, gotAlerts) {
+					var testName string
+					if tg.TestGroupName != "" {
+						testName = fmt.Sprintf("    name: %s,\n", tg.TestGroupName)
 					}
+					expString := indentLines(expAlerts.String(), "            ")
+					gotString := indentLines(gotAlerts.String(), "            ")
+					errs = append(errs, errors.Errorf("%s    alertname:%s, time:%s, \n        exp:%v, \n        got:%v",
```

Comment:
Thos 
```suggestion
					errs = append(errs, errors.Errorf("%s    alertname: %s, time: %s, \n        exp:%v, \n        got:%v",
```

Since name is on its own line, we could move time to its own line and remove the trailing commas